src/promptflow-tracing/promptflow/tracing/_span_enricher.py

```diff
@@ -0,0 +1,45 @@
+from typing import Dict
+
+from .contracts.trace import TraceType
+
+
+class SpanEnricher:
+    def __init__(self):
+        pass
+
+    def enrich(self, span, inputs, output):
+        from ._trace import enrich_span_with_output
+
+        enrich_span_with_output(span, output)
+
+
+class SpanEnricherManager:
+    _instance = None
+
+    def __init__(self):
+        self._type2enricher: Dict[str, SpanEnricher] = {}
+        self._base_enricher = SpanEnricher()
+
+    @classmethod
+    def get_instance(cls) -> "SpanEnricherManager":
+        if cls._instance is None:
+            cls._instance = SpanEnricherManager()
```

Comment:
Should we add lock here?

Reply:
Actually such instance would be initialized while import the code, so currently we don't need to worry about this.